fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricLanguageProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.provider;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.google.gson.JsonObject;
+
+import net.fabricmc.fabric.api.datagen.v1.provider.consumers.TranslationConsumer;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.DataWriter;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+
+/**
+ * Extend this class and implement {@link FabricLanguageProvider#generateTranslations(TranslationConsumer)}.
+ * Make sure to use {@link FabricLanguageProvider#FabricLanguageProvider(FabricDataGenerator, String)} FabricLanguageProvider} to declare what language code is being generated if it isn't {@code en_us}.
+ *
+ * <p>Register an instance of the class with {@link FabricDataGenerator#addProvider} in a {@link net.fabricmc.fabric.api.datagen.v1.DataGeneratorEntrypoint}
+ */
+public abstract class FabricLanguageProvider implements DataProvider {
+	protected final FabricDataGenerator dataGenerator;
+	private final String languageCode;
```

Comment:
Nitpick: Are we happy with just a String here or would it be better as a [java.util.Locale](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/Locale.html) Im unsure if there is any benefit

Reply:
Are all the Minecraft languages even valid locales (like upside-down or Shakespeare English)?

Reply:
Oh good point, doesnt seem like MC uses it, so might be best left as a string 👍